### DISCUSSION TITLE
tailwind with next and mui

### DISCUSSION THREAD

**Discussion by cadentic** [vercel/next.js#40493] [Help]
hi

my integration with nextjs mui 5 and tailwind css never works.
any guideline with mui next and tailwind will be appreciated 

[nextjs with tailwind and mui](https://github.com/cadentic/strapi-webdev/tree/main/zettadaten_fe_next17)

**Comment by DaRabus** [comment]
Mixing 2 different style providers isn't usually that good idea.

Why do you want to do that?

**Reply by cadentic** [reply]
bcz people has their own choices and existing works wouldn't go out of the window. 
moreover that it is possible to use both of them together in fact mui doesn't claim it is not supported. apart from that, it is a little more flexible to use both of them together to come up with minute perfections.

**Reply by cadentic** [reply]
tailwind with style component can be used together with twin macro it's nothing wrong i guess.

**Comment by rafde** [comment]
@cadentic I've been using MUI with tailwind for a while and haven't had issues. All tailwind classes I pass are to `className`. If you expect emotion to use tailwind classes, that won't work.

For example,
https://github.com/cadentic/strapi-webdev/blob/main/zettadaten_fe_next17/components/Navigation.js#L144
```jsx
<AppBar position="sticky" bgcolor="Colors.brown" sx={{ top: 'auto', bottom: 2 , px: 'auto', color: '#3a3632' , fontcolor: '#faf8f7'}}>
```
won't use tailwind classes. You would have to turn this into
```jsx
{/* I don't know what  color: '#3a3632' is for, so I excluded it for this example*/}
<AppBar className="sticky top-auto bottom-[2px] color-[#faf8f7]" bgcolor="Colors.brown">
```
